File: volume 119/11934.py
while True:
    arr=list(map(int,input().split()))
    if(arr[0]==0 and arr[1]==0 and arr[2]==0 and arr[3]==0 and arr[4]==0):
        break
    cnt=0
    for i in range(0,arr[4]+1):
        res=arr[0]*i*i+arr[1]*i+arr[2]
        if(res%arr[3]==0):
            cnt+=1
    print(cnt)

File: volume 119/11934.cpp
#include<iostream>
using namespace std;
int main(){
    int a,b,c,d,L;
    while(cin>>a>>b>>c>>d>>L){
        if(a==0 && b==0 && c==0 && d==0 && L==0) break;
        int cnt=0;
        for(int i=0;i<=L;i++){
            int res=a*i*i+b*i+c;
            if(res%d==0) cnt++;
        }
        cout<<cnt<<endl;
    }
}